pam/integration-tests/gdm-integration_test.go

```diff
@@ -0,0 +1,389 @@
+						Challenge: "goodpass",
+					}),
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Wait{}),
+				},
+			},
+			wantPamErrorMessages: []string{
+				fido1AuthID + " should have wait set to true",
+			},
+			wantError:       pam.ErrAuth,
+			wantAcctMgmtErr: ignoreError,
+		},
+	}
+	for name, tc := range testCases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Cleanup(pam_test.MaybeDoLeakCheck)
+			t.Parallel()
+
+			// We run a daemon for each test, because here we don't want to
+			// make assumptions whether the state of the broker and each test
+			// should run in parallel and work the same way in any order is ran.
+			ctx, cancel := context.WithCancel(context.Background())
+			socketPath, stopped := testutils.RunDaemon(ctx, t, daemonPath,
+				testutils.WithEnvironment(grouptests.GPasswdMockEnv(t, gpasswdOutput, groupsFile)...),
+			)
+			t.Cleanup(func() {
+				cancel()
+				<-stopped
+			})
+			serviceFile := createServiceFile(t, "module-loader", libPath,
+				[]string{"socket=" + socketPath}, pamDebugIgnoreError)
+
+			gh := newGdmTestModuleHandler(t, serviceFile, tc.pamUser)
+			t.Cleanup(func() { require.NoError(t, gh.tx.End()) })
+			gh.eventPollResponses = tc.eventPollResponses
+
+			if tc.supportedLayouts == nil {
+				gh.supportedLayouts = []*authd.UILayout{pam_test.FormUILayout()}
+			}
+
+			gh.protoVersion = gdm.ProtoVersion
+			if tc.protoVersion != 0 {
+				gh.protoVersion = tc.protoVersion
+			}
+
+			gh.selectedBrokerName = tc.brokerName
+			if gh.selectedBrokerName == "" {
+				gh.selectedBrokerName = exampleBrokerName
+			}
+
+			gh.selectedAuthModeIDs = tc.authModeIDs
+			if gh.selectedAuthModeIDs == nil {
+				gh.selectedAuthModeIDs = []string{passwordAuthID}
+			}
+
+			authResult := make(chan error)
+			go func() {
+				authResult <- gh.tx.Authenticate(pam.Flags(0))
+			}()
+
+			var err error
+			select {
+			case <-time.After(10 * time.Second):
+				t.Fatal("Authentication timed out!")
+			case err = <-authResult:
+			}
+
+			require.ErrorIs(t, err, tc.wantError)
+			require.Equal(t, tc.wantPamErrorMessages, gh.pamErrorMessages,
+				"PAM Error messages do not match")
+			require.Equal(t, tc.wantPamInfoMessages, gh.pamInfoMessages,
+				"PAM Info messages do not match")
+
+			require.ErrorIs(t, gh.tx.AcctMgmt(pam.Flags(0)), tc.wantAcctMgmtErr,
+				"Account Management PAM Error messages do not match")
+
+			if tc.wantError != nil {
+				return
+			}
+
+			user, err := gh.tx.GetItem(pam.User)
+			require.NoError(t, err)
+			require.Equal(t, tc.pamUser, user)
+		})
+	}
+}
+
+func buildPAMLibrary(t *testing.T) string {
+	t.Helper()
+
+	cmd := exec.Command("go", "build", "-C", "..",
+		"-buildmode=c-shared", "-gcflags=-dwarflocationlists=true")
+	cmd.Env = append(os.Environ(), `CGO_CFLAGS=-O0 -g3`)
+	if testutils.CoverDir() != "" {
+		// -cover is a "positional flag", so it needs to come right after the "build" command.
+		cmd.Args = append(cmd.Args, "-cover")
+	}
```

Comment:
The comment is very specific that it needs to come `right after the "build" command`. Did you check if we are getting coverage by appending it after all the other flags? (You should check the coverage of only running `TestGdmModule` to see if it's working).

Reply:
I've moved it as first, it shouldn't change much though since we're testing a library that we dlload through a C library, so... Go gets lost on that.